src/runtime.rs

```diff
@@ -175,6 +175,11 @@ impl Runtime {
         let dg = self.dependency_graph.lock();
         let thread_id = thread::current().id();
 
+        // Cycle in the same thread.
+        if thread_id == other_id {
```

Comment:
Do we even need to acquire the graph lock here?

Reply:
I don't think so